vignettes/tcplfit2-vignette.Rmd

```diff
@@ -161,3 +166,159 @@ The input data for this example contains 6 signatures for one chemical in a tran
+## Order by the following columns
+setkeyv(dat, c('acid', 'srcf', 'apid', 'coli', 'rowi', 'spid', 'conc'))
+
+## Define replicate id (rpid) column for test compound wells
+nconc <- dat[wllt == "t" , ## denotes test well as the well type (wllt)
+             list(n = lu(conc)),
+             by = list(acid, apid, spid)][ , list(nconc = min(n)), by = acid]
+dat[wllt == "t" & acid %in% nconc[nconc > 1, acid],
+    rpid := paste(acid, spid, wllt, srcf, apid, "rep1", conc, sep = "_")]
+dat[wllt == "t" & acid %in% nconc[nconc == 1, acid],
+    rpid := paste(acid, spid, wllt, srcf, "rep1", conc, sep = "_")]
+
+## Define rpid column for non-test compound wells
+dat[wllt != "t",
+    rpid := paste(acid, spid, wllt, srcf, apid, "rep1", conc, sep = "_")]
+
+## set repid based on rowid
+dat[, dat_rpid := rowid(rpid)]
+dat[, rpid := sub("_rep[0-9]+.*", "",rpid, useBytes = TRUE)]
+dat[, rpid := paste0(rpid,"_rep",dat_rpid)]
+
+# Define concentration index
+indexfunc <- function(x) as.integer(rank(unique(x))[match(x, unique(x))])
+dat[ , cndx := indexfunc(conc), by = list(rpid)]
+```
+
+Levels 2 and 3 are used for data adjustments and normalization. 
+
+```{r example4_mc2, fig.height = 6, fig.width = 7}
+# If no adjustments are required for the data, the corrected value (cval) should be set as original rval
+dat[,cval := rval]
+
+## Poor well quality (wllq) wells should be removed
+dat <- dat[!wllq == 0,]
+
+## Fitting generally cannot occur if response values are NA therefore values need to be removed
+dat <- dat[!is.na(cval),]
+
+## A column for log10 concentration is added as some of the mc3 methods require logc. Given logging concentration, conc=0 are not allowed therefore a dummy non-zero value should be used
+dat[conc == 0 , conc := 0.01]
+dat[ , logc := log10(conc)]
+
+#As a final step to prepare the dataset tcplfit2 processing, a dummy aeid is required if using mc3_mthds from tcpl
+dummy_aeid <- 99999
+dat[,aeid := dummy_aeid]
+
+## Set aeid as a key
+setkey(dat,aeid)
+```
+
+Once the data is initialized to a point where the required fields are available, the methods included in the `tcpl` package can be identifued and applied without the need for a database connection. You can see the list of available methods for Level 3 in the table below:
+
+```{r example4_mthdlist, fig.height = 6, fig.width = 7, warning = FALSE, echo = FALSE}
+mthd_funcs <- tcpl:::mc3_mthds()
+DT::datatable(tcpl::tcplMthdList(3),rownames= FALSE, options = list(scrollX = T))
+```
+
+Here three normalization methods are selected and applied to the data.  Note because of the way `tcpl` handles the application of functions, the dataframe must be called `dat`. In the future, `tcpl` will export these functions so that they can be applied to any dataset without the need for a specific name or dummy aeid.
+
+```{r example4_mc3methods, fig.height = 6, fig.width = 7, results = 'hide'}
+# apply level 3 methods
+lapply(mthd_funcs[["bval.apid.nwlls.med"]](dummy_aeid), eval)
```

Comment:
Are these lines of code directly applying the normalization steps from tcpl to the 'dat' object (without the DB connection)?  If yes, (a) pretty cool, and (b) I would just make sure to mention that explicitly in the note above or better clarify that with comments here 'in-line' with the code chunk.

I would also suggest adding 'in-line' comments to the code with a brief note of what these normalization steps are.